sycl/doc/extensions/proposed/sycl_ext_intel_virtual_functions.asciidoc

```diff
@@ -0,0 +1,579 @@
+`sycl_ext_oneapi_named_sub_group_sizes` extension, virtual member functions can
+only be called from kernels which *don't* have `reqd_sub_group_size` attribute
+set on them explicitly, or otherwise behavior is undefined.
+
+=== Kernel bundles and device images
+
+When an object of a polymorphic class is constructed, it stores a pointer to
+virtual table, which points to its virtual member functions. Addresses of those
+functions are accessible and valid only within a kernel bundle containing a
+kernel which used to construct an object.
+
+Performing calls to virtual member functions of an object constructed in a
+kernel from a different kernel bundle is an undefined behavior.
+
+[source,dpcpp]
+----
+using syclext = sycl::ext::oneapi::experimental;
+
+struct Base {
+  virtual SYCL_EXT_ONEAPI_INDIRECTLY_CALLABLE_PROPERTY() void foo() {}
+};
+
+class Constructor;
+class Use;
+
+int main() {
+  sycl::queue Q;
+
+  Base *Obj = sycl::malloc_device<Base>(1, Q);
+  int *Result = sycl::malloc_shared<int>(2, Q);
+
+  auto bundleA
+    = sycl::get_kernel_bundle<sycl::bundle_state::executable>(Q.get_context(),
+        {sycl::get_kernel_id<Constructor>()});
+  auto bundleB
+    = sycl::get_kernel_bundle<sycl::bundle_state::executable>(Q.get_context(),
+        {sycl::get_kernel_id<Use>()});
+
+
+  Q.submit([&](sycl::handler &CGH) {
+    CGH.use_kernel_bundle(bundleA);
+    CGH.single_task<Constructor>([=] {
+      // Only placement new can be used within device functions.
+      new (Obj) Derived;
+    });
+  });
+
+  Q.submit([&](sycl::handler &CGH) {
+    CGH.use_kernel_bundle(bundleB);
+    CGH.single_task<Use>(syclext::properties{syclext::calls_indirectly<>}, [=] {
+      // Call to 'Base::foo' is an undefined behavior here, because 'Obj' was
+      // constructed within kernel bundle `bundleA`
+      Obj->foo();
+    });
+  });
+
+  return 0;
+}
+----
+
+== Example usage
+
+[source,dpcpp]
+----
+#include <sycl/sycl.hpp>
+
+using syclext = sycl::ext::oneapi::experimental;
+
+class Base {
+public:
+  virtual SYCL_EXT_ONEAPI_INDIRECTLY_CALLABLE_PROPERTY()
+  int get_random_number() {
+    return 4; // Chosen by fair dice roll. Guaranteed to be random
+  }
+
+  // Not considered to be a device function, can use full set of C++ features
+  virtual int get_host_random_number() {
+    throw std::runtime_error("Not Implemented");
+  }
+};
+
+class Derived : public Base {
+public:
+  SYCL_EXT_ONEAPI_INDIRECTLY_CALLABLE_PROPERTY()
+  int get_random_number() override {
+    return 221;
+  }
+};
+
+int main() {
+  sycl::queue Q;
+
+  Base *Obj = sycl::malloc_device<Derived>(1, Q);
+  int *Result = sycl::malloc_shared<int>(1, Q);
+
+  Q.single_task([=] {
+    // Only placement new can be used within device functions.
+    new (Obj) Derived;
+  });
+
+  auto props = syclext::properties{syclext::calls_indirectly<>};
+  Q.single_task(props, [=] {
+    Base B;
+    Result[0] = B.get_random_number();
+  }).wait();
+  assert(A[0] == 4);
```

Comment:
```suggestion
  assert(Result[0] == 4);
```

There is no variable `A`, so I think this is what you intended?  Same for the other occurrences of `A` below.